client/javascripts/directives.js

```diff
@@ -0,0 +1,201 @@
+// Display Pie Chart of Macronutrients
+// Following tutorial from 
+// http://zeroviscosity.com/d3-js-step-by-step/step-1-a-basic-pie-chart
+app.directive("pie", function(){
+  return {
+    templateUrl: "/client/partials/pie.html",
+    restrict: 'AE',
+    scope: {
+      dataset: '<',
+      pieType: '<'
+    },
+    controller: function(){},
+    controllerAs: 'vm',
+    bindToController: true,
+    link: function(scope, element, attrs, vm) {
+
+      var dataset = vm.dataset;
+
+      var colors;
+      var macroColors = ['#ECD078', '#D95B43', '#C02942', '#542437', '#53777A', '#CFF09E', '#A8DBA8', '#3B8686'];
+
+      if (vm.pieType === "macroPie") {
+        colors = macroColors;
+      } else {
+        // Use a different color pallet in the future.
+        colors = macroColors;
+      }
+
+      var color = d3.scale.ordinal().range(colors); 
+
+       var height = 360;
+       var width = 360;
+       var radius = Math.min(width, height) / 2;
+       var donutWidth = 100;
+       
+       dataset.forEach(function(d) {
+         d.enabled = true;
+       });
+
+       // Creates a canvas and sets placeholder values
+       var svg = d3.select('div')
```

Comment:
I don't think this is actually scoped to the template of the directive as far as I can see - so you'll just attach the d3 chart to the first div on the page.

Try using `d3.select(element)`